know/rdf/parser/wikidata_parser.py

```diff
@@ -1,26 +1,44 @@
 # pylint: disable=no-self-use,unused-import,
 import re
+from qwikidata.sparql import return_sparql_query_results
 from rdf.parser.abstract_parser import AbstractParser, EntityType
+from rdf.parser.sparql_reader import read_sparql
 class WikidataParser(AbstractParser):
     """parses wikidata sources"""
 
     def __init__(self, uri):
         super().__init__(uri)
-        self.entity_id = re.search('Q[0-9]+', self.uri)
+        self.entity_id = re.search('Q[0-9]+', self.uri).group(0)
```

Comment:
Change this to some form of `get_wikidata_id_from_uri` function (or some better name), call it here, then add basic tests for it just to ensure it works.